Source/WebCore/dom/ScriptElement.cpp

```diff
@@ -93,6 +94,15 @@ void ScriptElement::childrenChanged(const ContainerNode::ChildChange& childChang
 {
     if (m_parserInserted == ParserInserted::No && childChange.isInsertion() && element().isConnected())
         prepareScript(); // FIXME: Provide a real starting line number here.
+
+    if (childChange.source == ContainerNode::ChildChange::Source::API)
+        m_childrenChangedByApi = true;
```

Comment:
This doesn't actually seem to get triggered by the document.write test which is a bit bizarre?